### ISSUE TITLE
Config file is ignored when run ruff from the plugin

### ISSUE THREAD

**Issue by koxudaxi** [koxudaxi/ruff-pycharm-plugin#291]
**Describe the bug**
Several users have pointed out a problem with config information in `pyproject.toml` and `ruff.toml` not being read.
However, the config information is applied when the `ruff` command is invoked directly from the terminal.

I could not reproduce the problem in my environment, so I have created this issue to sort out and resolve the issue.

**To Reproduce**
1. Add configs as `pyproject.toml` or `ruff.toml` in a project.
2. Write code that violates the rules.
3. See wrong warning

**Expected behavior**
The plugin doesn't show wrong waring (respect configs)

**Screenshots**
I can't reproduce the problem

**Environments (please complete the following information):**
 - IDE: [e.g. PyCharm Community 2022.1 ]
 - OS: [e.g. macOS  12.2.0 ]
 - Ruff Version [e.g. 1.9.0 ]
 - Plugin version [e.g. 0.3.11 ]

**Related Isses**
https://github.com/koxudaxi/ruff-pycharm-plugin/issues/277
https://plugins.jetbrains.com/plugin/20574-ruff/reviews#review=91176-91183
https://plugins.jetbrains.com/plugin/20574-ruff/reviews#review=90927-91184

**Comment by koxudaxi** [comment]
@xyxz-web 
Thank you for sharing the example.
I have tried the config.
But, It works fine in my local.

Could you please share the plugin version, ruff version, and ruff config? 

my environment are
IDE: PyCharm 2023.2.1 (Professional Edition)
OS: macOS 14
Ruff Version 0.1.1
Plugin version 0.0.24

**Comment by koxudaxi** [comment]
> I did upgrade several dependencies

After the dependencies are installed, and PyCharm has re-detected the package, the plugin will re-set the path to the ruff.

> restart PyCharm after activating the ruff-lsp option.

ruff-lsp is not enabled before restarting PyCharm.
Should we show a popup dialog to restart? :thinking:

**Comment by ttrei** [comment]
I think the `pyproject.toml` discovery differs between the plugin and running `ruff` via command-line.
https://docs.astral.sh/ruff/configuration/#pyprojecttoml-discovery
(Note that passing the configuration file explicitly via `--config` disables the intelligent auto-discovery of the closest configuration file to the subject python file)

I observed the subject issue within a Pycharm project where the content root doesn't contain the .idea directory. Something like this:
```
~/PycharmProjects/foo-project/.idea/
~/dev/foo/ # this is the content root
~/dev/foo/.venv/ # virtual environment with ruff binary used by the Ruff plugin
```
"Ruff config file" was left empty in the plugin configuration.
I placed `ruff.toml` both in `~/PycharmProjects/foo-project/` and in `~/dev/foo/`, and tried to format some python files in `~/dev/foo/`:
- Ruff plugin used `~/PycharmProjects/foo-project/ruff.toml`
- `ruff` binary from command-line used `~/dev/foo/ruff.toml`, regardless of current working directory

**Comment by kamilos956** [comment]
I have the same issue. In the `pyproject.toml` file in the section `tool.ruff.lint.per-file-ignores` I have an entry `"__init__.py" = ["PLC0414"]`, which in the Pycharm show me an error `PLC0414`. But when I run ruff with CLI everything is ok. Is it any update to this issue?

**Comment by bradenneal1** [comment]
I am also having the same issue as above i.e. `[tool.ruff.lint.per-file-ignores]` rules are being ignored from `pyproject.toml` when using the PyCharm plugin, where other settings are working correctly e.g. `[tool.ruff.lint]`.

Running `ruff` via the command-line detects the `[tool.ruff.lint.per-file-ignores]` settings and hence the issue appears isolated to (1) the ruff-plugin and (2) `per-file-ignores` settings specifically.

Environment:
OS: RHEL9
PyCharm: 2024.2.3 (Community Edition)
ruff version: ruff 0.6.9
ruff-plugin: 0.0.41

**Comment by koxudaxi** [comment]
I think the issue has been fixed.
